carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,260 @@
+
+        """
+        output = df.copy()
+
+        # Normalization
+        output[self._continuous] = self._scaler.transform(output[self._continuous])
+
+        # Encoding
+        output[self._encodings] = 0
+        for encoding in self._encodings:
+            for cat in self._categoricals:
+                if cat in encoding:
+                    value = encoding.split(cat + "_")[-1]
+                    output.loc[output[cat] == value, encoding] = 1
+                    break
+
+        # Get correct order
+        output = output[self._feature_input_order]
+
+        return output
+
+    def need_pipeline(self, x):
+        """
+        Checks if ML model input needs pipelining.
+        Only DataFrames can be used to pipeline input.
+
+        Parameters
+        ----------
+        x : pd.DataFrame or np.Array
+
+        Returns
+        -------
+        bool : Boolean
+            True if no pipelining process is already taken
+        """
+        if not isinstance(x, pd.DataFrame):
+            return False
+
+        if x.select_dtypes(exclude=[np.number]).empty:
+            return False
+
+        return True
+
+    @property
+    def name(self):
+        """
+        Contains meta information about the model, like name, dataset, structure, etc.
+
+        Returns
+        -------
+        name : str
+            Individual name of the ml model
+        """
+        return self._name
+
+    @property
+    def feature_input_order(self):
+        """
+        Saves the required order of feature as list.
+
+        Prevents confusion about correct order of input features in evaluation
+
+        Returns
+        -------
+        ordered_features : list of str
+            Correct order of input features for ml model
+        """
+        return self._feature_input_order
+
+    @property
+    def backend(self):
+        """
+        Describes the type of backend which is used for the ml model.
+
+        E.g., tensorflow, pytorch, sklearn, ...
+
+        Returns
+        -------
+        backend : str
+            Used framework
+        """
+        return self._backend
+
+    @property
+    def raw_model(self):
+        """
+        Returns the raw ml model built on its framework
+
+        Returns
+        -------
+        ml_model : tensorflow, pytorch, sklearn model type
+            Loaded model
+        """
+        return self._model
+
+    def predict(self, x):
+        """
+        One-dimensional prediction of ml model for an output interval of [0, 1]
+
+        Shape of input dimension has to be always two-dimensional (e.g., (1, m), (n, m))
+
+        Parameters
+        ----------
+        x : np.Array or pd.DataFrame
+            Tabular data of shape N x M (N number of instances, M number of features)
+
+        Returns
+        -------
+        output : np.ndarray
+            Ml model prediction for interval [0, 1] with shape N x 1
+        """
+
+        assert len(x.shape) == 2
```

Comment:
See above comments about asserts vs exceptions

Reply:
This should be solved in PR #17 and disappears with merging.